src/dfmt.d

```diff
@@ -115,8 +115,25 @@ private:
         assert (index < tokens.length);
         if (current.type == tok!"comment")
         {
+            const i = index;
+            if (i > 0)
+            {
+                if (tokens[i-1].line < tokens[i].line)
+                {
+                    if (tokens[i-1].type != tok!"comment"
+                        && tokens[i-1].type != tok!"{")
+                        newline();
+                }
+                else
+                    output.write(" ");
+            }
             writeToken();
-            newline();
+            if (i >= tokens.length-1)
+                newline();
+            else if (tokens[i+1].line > tokens[i].line)
+                newline();
+            else if (tokens[i+1].type != tok!"{")
+                output.write(" ");
```

Comment:
output.write doesn't increase the counter for the length of the current line the way that write() does. Is this intentional?